services/base.js

```diff
@@ -104,17 +112,30 @@ class BaseService {
    */
   static prepareExamples() {
     return this.examples.map(
-      ({ title, previewUrl, query, exampleUrl, documentation }) => {
-        if (!previewUrl) {
-          throw Error(`Example for ${this.name} is missing required previewUrl`)
+      ({
+        title,
+        staticExample,
+        previewUrl,
+        query,
+        exampleUrl,
+        documentation,
+      }) => {
+        if (!previewUrl && !staticExample) {
+          throw Error(
+            `Example for ${
+              this.name
+            } is missing required previewUrl or staticExample`
```

Comment:
Total aside: It would be helpful if instead of printing the classname, we printed the file where the class is defined. I wonder if we can do this by introspecting in the BaseService constructor.

Reply:
I did something similar, to a different end, in #1934.

Reply:
That's true, but I'm not sure we can use the same approach here. `caller()` returns `null` in this context

Reply:
Yea, that makes sense.

I was thinking there would be a way to snag it when the class was defined, though there’s no code from the superclass that runs at that time. The filename could be identified and associated with the service class at the time the service module is loaded. It’d be kind of dirty but might still be worth doing. I’ll give it some thought.